### PR TITLE
added husky

### PR DESCRIPTION
**Pull request by zarinaazamatova**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,9 +33,14 @@
...
+    "husky": "^8.0.3",
     "jest-environment-jsdom": "^29.5.0",
+    "lint-staged": "^13.2.2",
     "prettier": "^2.8.8",
     "ts-jest": "^29.1.0",
     "typescript": "^5.1.3"
-  }
+  },
+  "lint-staged": {
+    "**/*.{js,jsx,ts,tsx}": "eslint --fix --color"
```
Comment (Alex-2kZharkov) [inline package.json:44]: We need to modify this command. We not only want to check code with eslint but we also want to compile it from TypeScript to make sure it was a valid TS afterall. To do it, modify the `lint-staged` like this:
```
"lint-staged": {
  "**/*.{js,jsx,ts,tsx}": [
    "eslint --fix --color",
    "tsc --noEmit"
   ]
  }
```

**Review by Alex-2kZharkov — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (zarinaazamatova) [reply package.json]: done

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
```
[the remark this answers is not in the document]
Comment (zarinaazamatova) [reply .husky/pre-commit:1]: done

File: package.json [hunk 3]
```diff
@@ -9,7 +9,8 @@
     "lint": "next lint",
     "format": "npx prettier --write .",
     "test": "jest --watch",
```
[the remark this answers is not in the document]
Comment (zarinaazamatova) [reply package.json]: done

**Review by Alex-2kZharkov — APPROVED** [review]

**Pull request merged**